### PR TITLE
Use 'pagehide' for page termination by default.

### PR DESCRIPTION
**Pull request by wu-hui**
Fix: https://github.com/firebase/firebase-js-sdk/issues/4236

See: https://developers.google.com/web/updates/2018/07/page-lifecycle-api#legacy-lifecycle-apis-to-avoid

### CODE REVIEW

File: packages/firestore/src/util/types.ts [hunk 1]
```diff
@@ -59,6 +59,14 @@ export interface WindowLike {
   removeEventListener(type: string, listener: EventListener): void;
 }
 
+/**
+ * Returns the page termination event to listen to. 'pagehide' is recommended
+ * if available, it falls back to the less reliable 'unload'.
+ */
+export function terminationEvent(window: WindowLike | null): string {
+  return 'onpagehide' in window! ? 'pagehide' : 'unload';
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firestore/src/util/types.ts]: Good point, done.

**Review by schmidt-sebastian — COMMENTED** [review]
Please add changeset.

File: packages/firestore/test/util/test_platform.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export class FakeWindow implements WindowLike {
       case 'storage':
         this.storageListeners.push(listener);
         break;
-      case 'unload':
+      case 'pagehide':
       case 'visibilitychange':
         // The spec tests currently do not rely on `unload`/`visibilitychange`
```
Comment (0xdevalias) [inline packages/firestore/test/util/test_platform.ts:63]: [nit] Looks like this comment refers to the old ‘unload’ rather than ‘pagehide’

Comment (schmidt-sebastian) [reply packages/firestore/test/util/test_platform.ts:63]: Thanks for catching this. WIll fix soon.

**Pull request merged**